subprojects/tooling-api/src/testFixtures/groovy/org/gradle/integtests/tooling/fixture/ToolingApi.groovy

```diff
@@ -127,19 +135,47 @@ class ToolingApi implements TestRule {
         return DaemonLogsAnalyzer.newAnalyzer(getDaemonBaseDir(), dist.version.version)
     }
 
-    void withConnector(Closure cl) {
+    void withConnector(@DelegatesTo(GradleConnector) Closure cl) {
         connectorConfigurers << cl
     }
 
-    def <T> T withConnection(Closure<T> cl) {
-        GradleConnector connector = connector()
+    def <T> T withConnection(@DelegatesTo(ProjectConnection) Closure<T> cl) {
+        def connector = connector()
         withConnection(connector, cl)
     }
 
-    def <T> T withConnection(GradleConnector connector, Closure<T> cl) {
+    def <T> T withConnection(connector, @DelegatesTo(ProjectConnection) Closure<T> cl) {
         return withConnectionRaw(connector, cl)
     }
 
+    def <T> T loadToolingLeanModel(Class<T> modelClass, @DelegatesTo(ModelBuilder<T>) Closure configurator = {}) {
+        def result = withConnection {
+            def builder = it.model(modelClass)
+            builder.tap(configurator)
```

Comment:
🧐 
```suggestion
            configurator(builder)
```

Reply:
The `tap` changes the `this` whereas with the call you need to use `it` to refer to the builder.
Or is there something else I'm missing?

Reply:
You are right, thanks for the clarification.